### PR TITLE
[IOTDB-5418] Add metric for mods file

### PR DESCRIPTION
**Pull request by THUMarkLau**
See [IOTDB-5418](https://issues.apache.org/jira/browse/IOTDB-5418).

This PR add metrics for modification files.

Scene of modification file generation:
- Deletion
- Compaction finish(compaction modification converts to modification)

Scene of modification file elimination:
- Compaction finish
- SG is deleted.

image

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java [hunk 1]
```diff
@@ -74,6 +78,30 @@ public long getFileNum(boolean seq) {
...
+    return modFileSize.get();
+  }
+
+  public void increaseModFileNum(int num) {
+    modFileNum.addAndGet(num);
+  }
+
+  public void decreaseModFileNum(int num) {
+    modFileNum.addAndGet(num);
+  }
```
Comment (SpriCoder) [inline server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java:95]: Maybe `-num`

File: server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java [hunk 2]
```diff
@@ -32,6 +32,10 @@ public class TsFileMetricManager {
   private final AtomicInteger seqFileNum = new AtomicInteger(0);
   private final AtomicInteger unseqFileNum = new AtomicInteger(0);
 
+  private final AtomicInteger modFileNum = new AtomicInteger(0);
+
+  private final AtomicLong modFileSize = new AtomicLong(0);
```
Comment (SpriCoder) [inline server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java:37]: Consider whether the size may too big?

File: server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java [hunk 3]
```diff
@@ -184,6 +198,8 @@ private void unbindTsFileMetrics(AbstractMetricService metricService) {
         MetricType.AUTO_GAUGE, Metric.FILE_COUNT.toString(), Tag.NAME.toString(), "seq");
     metricService.remove(
         MetricType.AUTO_GAUGE, Metric.FILE_COUNT.toString(), Tag.NAME.toString(), "unseq");
+    metricService.remove(
+        MetricType.AUTO_GAUGE, Metric.FILE_COUNT.toString(), Tag.NAME.toString(), "mods");
```
Comment (SpriCoder) [inline server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java:202]: Please remove `file_size` related Metric here.

**Review by SpriCoder — CHANGES_REQUESTED** [review]
Good Job! Some minor problems, PTAL

**Review by SpriCoder — APPROVED** [review]
LGTM~

File: server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java (hunk 1 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java]: fix

File: server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java (hunk 2 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java:37]: It's AtomicLong, the size of mods files can be up to 8388608 TB.

File: server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java (hunk 3 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java:204]: fix

**Pull request merged**